frontend/src/AccountCreation/SignUpProfessor.tsx

```diff
@@ -4,26 +4,104 @@ import Stack from "react-bootstrap/Stack";
 import { Button } from "react-bootstrap";
 import Form from "react-bootstrap/Form";
 import Alert from "react-bootstrap/Alert";
-import React, { useState } from "react";
+import React, { FormEvent, useState } from "react";
 
 const SignUpProfessor = () => {
   const nav = useNavigate();
   const [error, setError] = useState("");
 
-  const submit: React.FormEventHandler<HTMLFormElement> = (
-    e: React.FormEvent<HTMLFormElement>
-  ) => {
+  const [name, setName] = useState("");
+  const [uid, setUID] = useState("");
+  const [courseName, setCourseName] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+
+  //check if passwords match
+  function validateForm(
+    name: string,
+    uid: string,
+    course_name: string,
+    password: string,
+    confirm_password: string
+  ) {
+    if (name.length === 0) {
+      setError("Please enter your name");
+      return false;
+    }
+    if (uid.length !== 9 || isNaN(parseInt(uid))) {
+      setError("Please correct your UID");
+      return false;
+    }
+    if (course_name.length === 0) {
+      setError("Please enter a course name");
+      return false;
+    } else if (password !== confirm_password) {
+      setError("Passwords do not match");
+      return false;
+    } else if (password.length < 8) {
+      setError("Password must be at least 8 characters long");
+      return false;
+    }
+    return true;
+  }
+
+  function handleProfCreation(e: FormEvent) {
     e.preventDefault();
-    setError((err) => (err ? "" : "Make Sure All Fields Are Filled"));
-    nav("/class");
-  };
+    if (!validateForm(name, uid, courseName, password, confirmPassword)) return;
+    fetch("http://localhost:8080/user", {
+      method: "POST",
+      mode: "cors",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name: name,
+        username: uid,
+        password: password,
+        type: "professor",
+      }),
+    })
+      //check for errors
+      .then((res) => {
+        if (res.status === 201) {
+          handleClassCreation(courseName);
+        } else {
+          setError("Error in Signing Up");
+        }
+      })
+      .catch(setError);
+  }
+
+  function handleClassCreation(courseName: string) {
+    fetch("http://localhost:8080/class", {
+      method: "POST",
+      mode: "cors",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name: courseName,
+      }),
+    })
+      //check for errors
+      .then((res) => {
+        if (res.status === 201) {
+          nav("/class");
+        } else {
+          setError("Error in Creating Class");
+        }
+      })
+      .catch(setError);
```

Comment:
Similarly to what was done in #49, you can throw a new error here `throw 'my error string'` and let the `.catch()` function call handle the fallthrough.

Reply:
Fixed